extensions/amp-live-list/0.1/live-list-manager.js

```diff
@@ -147,10 +151,16 @@ export class LiveListManager {
   fetchDocument_() {
     let url = this.url_;
     if (this.latestUpdateTime_ > 0) {
-      url = addParamToUrl(
+      url = addParamsToUrl(
         url,
-        'amp_latest_update_time',
-        String(this.latestUpdateTime_)
+        /** @type {JsonObject} */ ({
+          'amp_latest_update_time': String(this.latestUpdateTime_),
+          // AMP Caches do not always evict entries from their caches.
+          // A random number from the max safe range without BigInts helps add confidence the document is fresh.
+          'amp_random': String(
+            Math.floor(Math.random() * AMP_LIVE_LIST_MAX_RANDOM_NUMBER)
+          ),
```

Comment:
The cache will now have to fetch every time from the origin, and it'll cache that separately. Why not just make the cache always fetch when it's a live-list request (signified by `amp_latest_update_time`).

Note, too, that the origin will now see an `amp_random` query param. That has a chance to break origins that aren't expecting unknown queries.

Reply:
Changes to caches are not likely to happen quickly based on my understanding. Absolutely agree about the risk of these queries causing additional concern on origins.